cmd/release-controller/sync_analysis.go

```diff
@@ -6,21 +6,63 @@ import (
 	"k8s.io/klog"
 )
 
+func findStatusTag(is *imagev1.ImageStream, name string) *imagev1.TagEvent {
+	for i := range is.Status.Tags {
+		tag := &is.Status.Tags[i]
+		if tag.Tag == name {
+			if len(tag.Items) == 0 {
+				return nil
+			}
+			if len(tag.Conditions) > 0 {
+				if isTagEventConditionNotImported(tag) {
+					return nil
+				}
+			}
+			if specTag := findSpecTag(is.Spec.Tags, name); specTag != nil && (specTag.Generation == nil || *specTag.Generation > tag.Items[0].Generation) {
+				return nil
+			}
+			return &tag.Items[0]
+		}
+	}
+	return nil
+}
+
+// ensureAnalysisJobs creates and instantiates the specified number of occurrences of the release verification job for
+// a given release.  The jobs can be tracked via its labels and/or annotations:
+// Labels:
+//     "release.openshift.io/analysis": the name of the release tag (i.e. 4.9.0-0.nightly-2021-07-12-202251)
+// Annotations:
+//     "release.openshift.io/image": the SHA value of the release
+//     "release.openshift.io/dockerImageReference": the pull spec of the release
 func (c *Controller) ensureAnalysisJobs(release *Release, releaseTag *imagev1.TagReference) error {
+	statusTag := findStatusTag(release.Target, releaseTag.Name)
+	if statusTag == nil {
+		klog.V(2).Infof("Waiting for release %s to be imported before we can retrieve metadata", releaseTag.Name)
+		return nil
+	}
 	for name, analysisType := range release.Config.Analysis {
 		if analysisType.Disabled {
 			klog.V(2).Infof("Release %s analysis step %s is disabled, ignoring", releaseTag.Name, name)
 			continue
 		}
-		if analysisType.AnalysisJobCount == 0 {
+		if analysisType.AnalysisJobCount <= 0 {
 			klog.Warningf("Release %s analysis step %s configured without analysisJobCount, ignoring", releaseTag.Name, name)
 			continue
 		}
+		jobCount := analysisType.AnalysisJobCount
+		if jobCount > 20 {
+			jobCount = 20
```

Comment:
No.  On bad input, do nothing.  Warn as you're doing and skip the job.

People may not notice they're getting 20 instead of 30 jobs. They will notice (or should notice), that they get zero jobs instead of 30.